doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -11,92 +11,83 @@ These tactics use the equality :g:`eq:forall A:Type, A->A->Prop` defined in
-   resulting in instances `term`:sub:`1`' and `term`:sub:`2`' and then
-   replaces every occurrence of `term`:subscript:`1`' by `term`:subscript:`2`'.
-   Hence, some of the variables :g:`x`\ :sub:`i` are solved by unification,
-   and some of the types :g:`A`\ :sub:`1`:g:`, ..., A`\ :sub:`n` become new
-   subgoals.
+   :n:`forall (x__1: A__1) ... (x__n: A__n), eq term__1 term__2`
 
-   .. exn:: The @term provided does not end with an equation.
-      :undocumented:
-
-   .. exn:: Tactic generated a subgoal identical to the original goal. This happens if @term does not occur in the goal.
-      :undocumented:
-
-   .. tacv:: rewrite -> @term
-
-      Is equivalent to :n:`rewrite @term`
+   where :g:`eq` is the Leibniz equality or a registered setoid equality.
 
-   .. tacv:: rewrite <- @term
+   :n:`rewrite @one_term` finds subterms matching `term`\ :sub:`1` in the goal,
+   and replaces them with `term`\ :sub:`2` using the values bound by the match
+   (or the reverse if `<-` is given).
+   Some of the variables :g:`x`\ :sub:`i` are solved by unification,
+   and some of the types :n:`A__1, ..., A__n` may become new
+   subgoals.
 
-      Uses the equality :n:`@term`:sub:`1` :n:`= @term` :sub:`2` from right to left
+   Rewriting is
+   always done with setoid rewriting, even for Leibniz’s equality, so you must
+   ``Import Setoid`` to use this tactic.
 
-   .. tacv:: rewrite @term in @goal_occurrences
+   :n:`{+, @oriented_rewriter }`
+     The :n:`@oriented_rewriter`\s are applied sequentially
+     to the first goal generated by the previous :n:`@oriented_rewriter`.  If any of them fail,
+     the tactic fails.
 
-      Analogous to :n:`rewrite @term` but rewriting is done following
-      the clause :token:`goal_occurrences`. For instance:
+   :n:`@occurrences`
+     If :n:`@occurrences` specifies multiple occurrences, the tactic succeeds if
+     any of them can be rewritten.  If not specified, only the first occurrence
+     in the conclusion is replaced.
 
-      + :n:`rewrite H in H'` will rewrite `H` in the hypothesis
-        ``H'`` instead of the current goal.
-      + :n:`rewrite H in H' at 1, H'' at - 2 |- *` means
-        :n:`rewrite H; rewrite H in H' at 1; rewrite H in H'' at - 2.`
-        In particular a failure will happen if any of these three simpler tactics
-        fails.
-      + :n:`rewrite H in * |-` will do :n:`rewrite H in H'` for all hypotheses
-        :g:`H'` different from :g:`H`.
-        A success will happen as soon as at least one of these simpler tactics succeeds.
-      + :n:`rewrite H in *` is a combination of :n:`rewrite H` and :n:`rewrite H in * |-`
-        that succeeds if at least one of these two tactics succeeds.
+     .. todo PR above is correct?  rewrite ... at 1 2 3 works even if only 1 is rewritten?
 
-      Orientation :g:`->` or :g:`<-` can be inserted before the :token:`term` to rewrite.
+   :n:`@ltac_expr3`
+     If specified, is used to resolve all side conditions generated by the tactic.
```

Comment:
Should be moved below. We want to have finished describing the full syntax of `oriented_rewriter` first.

Reply:
I guess.  bfs order may work better for complex commands.